trin-core/src/portalnet/overlay_service.rs

```diff
@@ -410,6 +428,54 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
                         self.peers_to_ping.insert(node_id);
                     }
                 }
+                query_event = OverlayService::<TContentKey, TMetric>::query_event_poll(&mut self.queries) => {
+                    match query_event {
+                        // Send a FINDNODES on behalf of the query.
+                        QueryEvent::Waiting(query_id, node_id, request) => {
+
+                            // Look up the node's ENR.
+                            if let Some(enr) = self.node_enrs.read().get(&node_id) {
+
+                                let request = OverlayRequest::new(
+                                    request,
+                                    RequestDirection::Outgoing {
+                                        destination: enr.clone()
+                                    },
+                                    None,
+                                );
+
+                                self.active_query_requests.write().insert(node_id, query_id);
+
+                                debug!("[{:?}] Sending FINDNODES on behalf of query {}", self.protocol, query_id);
+                                let _ = self.request_tx.send(request);
+
+                            } else {
+                                error!("[{:?}] Query Pool wants to send a FINDNODES to {},
+                                        but ENR is not known.", self.protocol, node_id);
```

Comment:
I think we can make this more concise: "Unable to send query {query_id} to unknown ENR". Something like that.